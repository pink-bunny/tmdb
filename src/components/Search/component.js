import React from 'react';
import PropTypes from 'prop-types';
import { getIn } from 'formik';
import { Input, Typography } from 'antd';

const Search = ({
  placeholder,
  enterButton,
  field,
  onSearch,
  form: { touched, errors },
  ...props
}) => {
  const fieldTouched = getIn(touched, field.name);
  const fieldErrors = getIn(errors, field.name);

  return (
    <>
      <Input.Search
        placeholder={placeholder}
        size="large"
        enterButton={enterButton}
        onSearch={onSearch}
        onPressEnter={onSearch}
        {...field} /* eslint-disable-line */
        {...props} /* eslint-disable-line */
      />
      {fieldErrors && fieldTouched ? (
        <Typography.Paragraph
          type="danger"
          style={{ marginBottom: '16px' }}
        >
          {fieldErrors}
        </Typography.Paragraph>
      ) : null}
    </>
  );
};

Search.propTypes = {
  placeholder: PropTypes.string,
  enterButton: PropTypes.string,
  field: PropTypes.shape().isRequired,
  form: PropTypes.shape({
    touched: PropTypes.shape().isRequired,
    errors: PropTypes.shape().isRequired
  }).isRequired,
  onSearch: PropTypes.func.isRequired
};
Search.defaultProps = {
  placeholder: '',
  enterButton: 'Enter'
};

export default Search;
