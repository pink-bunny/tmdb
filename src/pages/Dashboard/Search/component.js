import React from 'react';
import PropTypes from 'prop-types';
import { Field } from 'formik';

import { Row, Col } from 'antd';
import SearchComponent from '../../../components/Search';

const Search = ({
  handleSubmit
}) => (
  <Row type="flex">
    <Col
      xs={{ span: 22, offset: 1 }}
      sm={{ span: 20, offset: 2 }}
      md={{ span: 18, offset: 3 }}
      lg={{ span: 16, offset: 4 }}
      xl={{ span: 14, offset: 5 }}
    >
      <div className="top-margin">
        <form>
          <Field
            name="search"
            placeholder="Enter movie name"
            enterButton="Search"
            onSearch={handleSubmit}
            component={SearchComponent}
          />
        </form>
      </div>
    </Col>
  </Row>
);

Search.propTypes = {
  handleSubmit: PropTypes.func.isRequired
};

export default Search;
