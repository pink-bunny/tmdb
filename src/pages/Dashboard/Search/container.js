import React from 'react';
import { withFormik } from 'formik';
import { connect } from 'react-redux';
import * as Yup from 'yup';

import Search from './component';

const SearchFormSchema = Yup.object().shape({
  search: Yup.string()
    .required('Search field can\'t be empty')
});

const SearchForm = withFormik({
  mapPropsToValues: () => ({ search: '' }),

  validationSchema: SearchFormSchema,

  handleSubmit: (values, { setSubmitting }) => {
    setTimeout(() => {
      alert(JSON.stringify(values, null, 2));
      setSubmitting(false);
    }, 1000);
  },

  displayName: 'SearchForm'
})(Search);

const mapDispatchToProps = {};

export default connect(null, mapDispatchToProps)(SearchForm);
