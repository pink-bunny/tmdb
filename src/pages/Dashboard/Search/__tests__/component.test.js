import React from 'react';
import { shallow } from 'enzyme';

import Search from '../component';

test('Search component matches snapshot', () => {
  const requiredProps = {
    handleSubmit: jest.fn()
  };
  const wrapper = shallow(<Search {...requiredProps} />);/* eslint-disable-line */

  expect(wrapper).toMatchSnapshot();
});
