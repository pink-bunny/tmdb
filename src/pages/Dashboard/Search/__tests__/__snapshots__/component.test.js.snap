// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component matches snapshot 1`] = `
<Row
  gutter={0}
  type="flex"
>
  <Col
    lg={
      Object {
        "offset": 4,
        "span": 16,
      }
    }
    md={
      Object {
        "offset": 3,
        "span": 18,
      }
    }
    sm={
      Object {
        "offset": 2,
        "span": 20,
      }
    }
    xl={
      Object {
        "offset": 5,
        "span": 14,
      }
    }
    xs={
      Object {
        "offset": 1,
        "span": 22,
      }
    }
  >
    <div
      className="top-margin"
    >
      <form>
        <Field
          component={[Function]}
          enterButton="Search"
          name="search"
          onSearch={[MockFunction]}
          placeholder="Enter movie name"
        />
      </form>
    </div>
  </Col>
</Row>
`;
